### PR TITLE
Improve a bit the project

### PR DESCRIPTION
**Pull request by uepoch**
This is only a start
actually it lacks a lot of logging

### CODE REVIEW

File: login.go [hunk 1]
```diff
@@ -49,74 +75,97 @@ func (b *backend) pathAuthLogin(ctx context.Context, req *logical.Request, d *fr
...
+						Period:       chefRole.Period,
+						Policies:     chefRole.VaultPolicies,
+						Metadata:     map[string]string{"role": chefRole.Name},
+						GroupAliases: []*logical.Alias{},
+					}
+					// n is usually between 1 or 5, it's ok to loop again
+					for _, r := range nodeRoles {
+						auth.GroupAliases = append(auth.GroupAliases, &logical.Alias{Name: "role" + r})
+					}
+					break
```
Comment (ShimmerGlass) [inline login.go]: You do not break out of the outer loop here. Is that intended ?

File: tools.go [hunk 2]
```diff
@@ -0,0 +1,34 @@
+package main
+
+import (
+	"time"
+)
+
+func maxDuration(xs ...time.Duration) time.Duration {
+	var max time.Duration
+	if len(xs) == 2 {
+		if xs[0].Nanoseconds() > xs[1].Nanoseconds() {
```
Comment (ShimmerGlass) [inline tools.go]: `time.Duration`s are comparable as-is

File: login.go [hunk 3]
```diff
@@ -49,74 +75,97 @@ func (b *backend) pathAuthLogin(ctx context.Context, req *logical.Request, d *fr
...
+					Period:       chefPolicy.Period,
+					Policies:     chefPolicy.VaultPolicies,
+					Metadata:     map[string]string{"policy": chefPolicy.Name},
+					GroupAliases: []*logical.Alias{
+						{
+							Name: "policy-" + chefPolicy.Name,
+						},
+					},
+				}
+			}
```
Comment (ShimmerGlass) [inline login.go:107]: no break out of here ?

File: login.go (hunk 1 above)
Comment (uepoch) [reply login.go]: nope

File: tools.go (hunk 2 above)
Comment (uepoch) [reply tools.go]: dead code, thanks, will remove

**Comment by ShimmerGlass** [comment]
:+1:

**Pull request merged**